barretenberg/cpp/pil/vm2/sha256.pil

```diff
@@ -0,0 +1,375 @@
+    pol commit w_2_rotr_19;
+    pol commit lhs_w_19;
+    pol commit rhs_w_19;
+    perform_round * (helper_w14 - (lhs_w_19 * 2**19 + rhs_w_19)) = 0;
+    perform_round * (w_2_rotr_19 - (rhs_w_19 * 2**13 + lhs_w_19)) = 0;
+    // w[i-2] `rightshift` 10
+    pol commit w_2_rshift_10;
+    pol commit lhs_w_10;
+    pol commit rhs_w_10;
+    perform_round * (helper_w14 - (lhs_w_10 * 2**10 + rhs_w_10)) = 0;
+    perform_round * (w_2_rshift_10 - lhs_w_10) = 0;
+    // s1 := (w[i-2] `rotr` 17) `xor` (w[i-2] `rotr` 19) `xor` (w[i-2] `rightshift` 10)
+    pol commit w_2_rotr_17_xor_w_2_rotr_19;
+    // #[LOOKUP_W_S_1_XOR_0]
+    // dummy_zero {w_2_rotr_17, w_2_rotr_19, w_2_rotr_17_xor_w_2_rotr_19, xor_sel}
+    // in
+    // binary.start {binary.acc_ia, binary.acc_ib, binary.acc_ic, binary.op_id};
+
+    pol commit w_s_1;
+    // #[LOOKUP_W_S_1_XOR_1]
+    // dummy_zero {w_2_rotr_17_xor_w_2_rotr_19, w_2_rshift_10, w_s_1, xor_sel}
+    // in
+    // binary.start {binary.acc_ia, binary.acc_ib, binary.acc_ic, binary.op_id};
+
+    // ========== START OF COMPRESSION BLOCK ==================
+
+    // ====== Computing S1 ====================
+    // e `rotr` 6
+    pol commit e_rotr_6;
+    pol commit lhs_e_6;
+    pol commit rhs_e_6;
+    perform_round * (e - (lhs_e_6 * 2**6 + rhs_e_6)) = 0;
+    perform_round * (e_rotr_6 - (rhs_e_6 * 2**26 + lhs_e_6)) = 0;
+    // e `rotr` 11
+    pol commit e_rotr_11;
+    pol commit lhs_e_11;
+    pol commit rhs_e_11;
+    perform_round * (e - (lhs_e_11 * 2**11 + rhs_e_11)) = 0;
+    perform_round * (e_rotr_11 - (rhs_e_11 * 2**21 + lhs_e_11)) = 0;
+    // e `rotr` 25
+    pol commit e_rotr_25;
+    pol commit lhs_e_25;
+    pol commit rhs_e_25;
+    perform_round * (e - (lhs_e_25 * 2**25 + rhs_e_25)) = 0;
+    perform_round * (e_rotr_25 - (rhs_e_25 * 2**7 + lhs_e_25)) = 0;
+
+    // pol S_1 = (E_0 `rotr` 6) `xor` (E_0 `rotr` 11) `xor` (E_0 `rotr` 25);
+
+    pol commit e_rotr_6_xor_e_rotr_11;
+    // #[LOOKUP_S_1_XOR_0]
+    // dummy_zero {e_rotr_6, e_rotr_11, e_rotr_6_xor_e_rotr_11, xor_sel}
+    // in
+    // binary.start {binary.acc_ia, binary.acc_ib, binary.acc_ic, binary.op_id};
+
+    pol commit s_1;
+    // #[LOOKUP_S_1_XOR_1]
+    // dummy_zero {e_rotr_6_xor_e_rotr_11, e_rotr_25, s_1, xor_sel}
+    // in
+    // binary.start {binary.acc_ia, binary.acc_ib, binary.acc_ic, binary.op_id};
+
+    // ==== COMPUTING CH ===========
+    // pol CH_0 = (E_0 `and` F_0) `xor` ((`not` E_0) `and` G_0);
+    pol commit e_and_f;
+
+    // #[LOOKUP_CH_AND_0]
+    // dummy_zero {e, f, e_and_f, and_sel}
+    // in
+    // binary.start {binary.acc_ia, binary.acc_ib, binary.acc_ic, binary.op_id};
+
+    pol commit not_e;
+    perform_round * (e + not_e - (2**32 - 1)) = 0;
+
+    pol commit not_e_and_g;
+    // #[LOOKUP_CH_AND_1]
+    // dummy_zero {not_e, g, not_e_and_g, and_sel}
+    // in
+    // binary.start {binary.acc_ia, binary.acc_ib, binary.acc_ic, binary.op_id};
+
+    pol commit ch;
+    // #[LOOKUP_CH_XOR]
```

Comment:
(not this one)
You can add lookups into precomputed I think. See how David did it.